testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,89 @@
+from assertpy import assert_that
+
+from pathlib import Path
+
+from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.executable import CustomScript, CustomScriptBuilder
+from lisa.testsuite import simple_requirement
+from lisa.util import commonlib
+from lisa.tools import Lscpu, Lsvmbus
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+    tags=[],
+)
+class gpu(TestSuite):
+
+    unsupported_distro = ["CLEARLINUX", "COREOS"]
+    grid_supported_sku = ["Standard_NV"]
+    cuda_supported_sku = ["Standard_NC", "Standard_ND"]
+        
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu drivers.
+
+            Steps:
+            1. Install the GPU drivers on the VM.
+            2. Verifies if gpu drivers can be loaded successfully.
+        """,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def gpu_validate_gpudriver(self, environment: Environment, node: Node) -> None:
+        # validate if distro is supported for GPU
+        if not validate_gpu_support(self, node):
+            raise SkippedException("distro not supported")
+
+        if any(map((node.AzureNodeSchema.vm_size).__contains__, grid_supported_sku)):
+            self.log.info(
+                f"The VM {node.AzureNodeSchema.vm_size}"
+                f" instance is with GRID device driver."
+            )
+            install_result = validate_grid_install(self, node)
+
+        elif any(map((node.AzureNodeSchema.vm_size).__contains__, cuda_supported_sku)):
+            self.log.info(
+                f"The VM {node.AzureNodeSchema.vm_size}"
+                f" instance is with CUDA device driver."
+            )
+            validate_cuda_install(self, node)
+
+        else:
+            raise SkippedException("Azure VM size not supported.")
+
+
+    # Helper functions
+
+    # TODO: validate the test params against GPU support test matrix
+    # when test matrix is available.
+    def validate_gpu_support(self, node: Node) -> bool:
```

Comment:
I thought it more, the GPU related operations should be a feature. So it can cross more distro and platforms. Refer to [document](https://github.com/microsoft/lisa/blob/main/docs/extension.md#feature) about how to define, implement, and use a feature.
The Gpu feature can include several methods, `is_supported`, `install_driver`, and `install_compute_sdk` (cuda or grid).